feat: add Ivy demo for Cysharp.ZString

MedhatZ opened the pull request:
## Summary
This PR adds a new Ivy demo application showcasing the usage of [Cysharp.ZString](https://github.com/Cysharp/ZString).

## Features
- Demonstrates `ZString.Concat`, `ZString.Format`, and `ZString.Join`
- Demonstrates usage of `ZString.CreateStringBuilder()`
- Demonstrates `ZString.PrepareUtf16<>`

## Structure
The example is located in the `zstring/` folder and includes:
- `HelloApp.cs` with interactive Ivy UI buttons
- `README.md` with instructions
- `assets/screenshot.png` as a visual reference

## Screenshot
![screenshot](zstring/assets/screenshot.png)

MedhatZ commented on zstring/Apps/HelloApp.cs:
```diff
@@ -0,0 +1,52 @@
+[App(icon: Icons.PartyPopper, title: "ZString Demo")]
+public class HelloApp : ViewBase
+{
+    public override object Build()   // ينفع تشيل ? لتسكّت التحذير
```
Fixed in latest commit. Thanks!

rorychatt reviewed (changes requested):
It is hard to understand what the library does.

My suggestion is to include also what is happening.

i.e. what strings we concatenate, what we build together, what was formatted

MedhatZ commented:
I've updated the demo based on the feedback:

- Added explicit **inputs and outputs** in the UI for Concat, Format, Join, StringBuilder, and Prepared Format.
- Updated `README.md` to document inputs/outputs with examples.
- Added a new screenshot (`assets/screenshot.png`) showing the results.

Please review and let me know if there’s anything else to adjust.


<img width="1917" height="975" alt="screenshot" src="https://github.com/user-attachments/assets/bb11c7c2-cb59-4274-8470-3defc042253a" />
<img width="1915" height="989" alt="screenshot2" src="https://github.com/user-attachments/assets/23459930-6ea9-4a5e-b6c8-8c9a0ee2fb40" />
